js-miniapp-sdk/src/miniapp.ts

```diff
@@ -35,6 +36,15 @@ interface MiniAppFeatures {
    * @returns The Promise of share info action state from injected side.
    */
   shareInfo(info: ShareInfoType): Promise<string>;
+
+  /**
+   * makes the http request at Host application layer.
+   *
+   * @param url requestUrl
+   * @param options requestOptions
+   * @returns success or failure http response.
+   */
+  fetch(url: string, options: MiniAppRequestInit): Promise<Response>;
```

Comment:
I think we should explicitly define the `Response` type too. Maybe it can be named `FetchResponse`?

Also, shouldn't `options` be optional? I wonder if we should use the same style as the browser's `fetch` where it can be called as any of the following:
- `fetch("https://example.com")`
- `fetch("https://example.com", {headers: headers})`
- `fetch({url: "https://example.com", headers: headers})`

What do you think?

Reply:
I agree with @corycaywood here that we should explicitly define the response interface.